plasmapy/diagnostics/tests/test_thomson.py

```diff
@@ -246,6 +247,72 @@ def test_spectral_density_minimal_arguments(single_species_collective_args):
     return (alpha, wavelengths, Skw)
 
 
+def test_invalid_input_all_three(single_species_collective_args):
```

Comment:
Good job understanding my convoluted test structures! These tests look good: my only comment is that you could probably use `pytest.parameterize` to cover all these cases in just one function. That would reduce the amount of duplicated testing code.

Reply:
I certainly agree, but I'll want to do a pass like that soon anyway, so don't worry too much about it here :)